wgengine/netstack/netstack.go

```diff
@@ -793,3 +796,56 @@ func stringifyTEI(tei stack.TransportEndpointID) string {
 func ipPortOfNetstackAddr(a tcpip.Address, port uint16) (ipp netaddr.IPPort, ok bool) {
 	return netaddr.FromStdAddr(net.IP(a), int(port), "") // TODO(bradfitz): can do without allocs
 }
+
+// removeECNBits modifies p to remove ECN bits, if any.
+func removeECNBits(p *packet.Parsed) {
+	if p.IPVersion == 0 {
+		return
+	}
+	if p.IPProto != ipproto.TCP {
+		// TODO(bradfitz): handle non-TCP too? for now only trying to
+		// fix the Issue 2642 problem.
+		return
+	}
+	if p.TCPFlags&(packet.TCPCWR|packet.TCPECNEcho) == 0 {
```

Comment:
Similar to TCPSynAck, I'd make a combined constant for this called something like TCPECN, so it's more obvious that you're masking ECN bits to someone who hasn't recently re-read that "ECN Echo" and "Congestion Window Reduced" are the two flags TCP ECN uses.

Reply:
done.